Realm/ObjectServerTests/SwiftMongoClientTests.swift

```diff
@@ -0,0 +1,1222 @@
+
+        let changeStream1: ChangeStream?
+        let changeStream2: ChangeStream?
+
+        if let queue = queue {
+            changeStream1 = collection.watch(filterIds: [objectIds[0]], delegate: watchTestUtility1, queue: queue)
+            changeStream2 = collection.watch(filterIds: [objectIds[1]], delegate: watchTestUtility2, queue: queue)
+        } else {
+            changeStream1 = collection.watch(filterIds: [objectIds[0]], delegate: watchTestUtility1)
+            changeStream2 = collection.watch(filterIds: [objectIds[1]], delegate: watchTestUtility2)
+        }
+
+        let teardownEx = expectation(description: "All changes complete")
+        DispatchQueue.global().async {
+            watchTestUtility1.isOpenSemaphore.wait()
+            watchTestUtility2.isOpenSemaphore.wait()
+            for i in 0..<3 {
+                let name: AnyBSON = .string("fido-\(i)")
+                collection.updateOneDocument(filter: ["_id": AnyBSON.objectId(objectIds[0])],
+                                             update: ["name": name, "breed": "king charles"]) { result in
+                    if case .failure = result {
+                        XCTFail("Should update")
+                    }
+                }
+                collection.updateOneDocument(filter: ["_id": AnyBSON.objectId(objectIds[1])],
+                                             update: ["name": name, "breed": "king charles"]) { result in
+                    if case .failure = result {
+                        XCTFail("Should update")
+                    }
+                }
+                watchTestUtility1.semaphore.wait()
+                watchTestUtility2.semaphore.wait()
+            }
+            changeStream1?.close()
+            changeStream2?.close()
+            teardownEx.fulfill()
+        }
+        wait(for: [watchEx, teardownEx], timeout: 60.0)
+    }
+
+    func testShouldNotDeleteOnMigrationWithSync() {
+        let user = try! logInUser(for: basicCredentials())
+        var configuration = user.configuration(testName: appId)
+
+        assertThrows(configuration.deleteRealmIfMigrationNeeded = true,
+                     reason: "Cannot set 'deleteRealmIfMigrationNeeded' when sync is enabled ('syncConfig' is set).")
+
+        var localConfiguration = Realm.Configuration.defaultConfiguration
+        assertSucceeds {
+            localConfiguration.deleteRealmIfMigrationNeeded = true
+        }
+    }
+}
+
+#if swift(>=5.5)
+@available(macOS 12.0, *)
+class AsyncAwaitMongoClientTests: SwiftSyncTestCase {
+    func setupMongoCollection() async throws -> MongoCollection {
+        let email = "realm_tests_do_autoverify\(randomString(7))@\(randomString(7)).com"
+        let password = randomString(10)
+        try await app.emailPasswordAuth.registerUser(email: email, password: password)
+
+        let credentials = Credentials.emailPassword(email: email, password: password)
+        let user = try await app.login(credentials: credentials)
```

Comment:
This should use our convenience method to create and login a user, which will eliminate the above 6 lines of code.